docs/openshift/deploy.md

```diff
@@ -0,0 +1,113 @@
+```
+$ az account list -o json
+[
+  {
+    "cloudName": "AzureCloud",
+    "id": "5eca53b6-18b4-4d9b-a4d4-a45a1ff367c8",
+    "isDefault": true,
+    "name": "Acme Corp",
+    "state": "Enabled",
+    "tenantId": "5a27b61b-1b6e-4be5-aa9d-0d5696076bb9",
+    "user": {
+      "name": "a2ada0c0-1a4d-4923-848f-c04b8b301c13",
+      "type": "servicePrincipal"
+    }
+  }
+]
+```
+
+## Deploy
+
+For this example, the subscription id is `5eca53b6-18b4-4d9b-a4d4-a45a1ff367c8`, the tenant id is `5a27b61b-1b6e-4be5-aa9d-0d5696076bb9`, the DNS prefix and a resource group is `openshift-red`, and location is `eastus`.
+
+Before running the `acs-engine deploy` command, you must fill in all missing fields in the `examples/openshift.json` file. See the ["Long Way" section below](#acs-engine-the-long-way) for the description of required values.
+
+Now you can run `acs-engine deploy` with the appropriate arguments:
+
+```
+$ acs-engine deploy --subscription-id 5eca53b6-18b4-4d9b-a4d4-a45a1ff367c8 \
+    --resource-group openshift-red --location eastus \
+    --api-model examples/openshift.json
+
+INFO[0034] Starting ARM Deployment (openshift-red-1843927849). This will take some time...
+INFO[0393] Finished ARM Deployment (openshift-red-1843927849).
+```
+
+`acs-engine` will output Azure Resource Manager (ARM) templates, SSH keys and a node configuration in `_output/openshift-red` directory:
+
+   * `_output/openshift-red/apimodel.json`
+   * `_output/openshift-red/azuredeploy.json`
+   * `_output/openshift-red/azuredeploy.parameters.json`
+   * `_output/openshift-red/master.tar.gz`
+   * `_output/openshift-red/node.tar.gz`
+
+Administrative note: By default, the directory where acs-engine stores cluster configuration (`_output/openshift-red` above) won't be overwritten as a result of subsequent attempts to deploy a cluster using the same `--dns-prefix`) To re-use the same resource group name repeatedly, include the `--force-overwrite` command line option with your `acs-engine deploy` command.
+
+Bonus tip: include an `--auto-suffix` option to append a randomly generated suffix to the dns-prefix to form the resource group name, for example if your workflow requires a common prefix across multiple cluster deployments. Using the `--auto-suffix` pattern appends a compressed timestamp to ensure a unique cluster name (and thus ensure that each deployment's configuration artifacts will be stored locally under a discrete `_output/<resource-group-name>/` directory).
+
+## ACS Engine the Long Way
+
+### Step 1: Generate an SSH Key
+
+In addition to using OpenShift APIs to interact with the clusters, cluster operators may access the master machine using SSH.
+
+If you don't have an SSH key [cluster operators may generate a new one](../ssh.md#ssh-key-generation).
+
+### Step 2: Create a Service Principal
+
+acs-engine needs a Service Principal to interact with Azure Resource Manager (ARM). Follow the instructions to [create a new service principal](../serviceprincipal.md).
```

Comment:
The OpenShift cluster needs a Service Principal ... , for example to dynamically create Azure persistent storage volumes.

Reply:
These instructions won't work if AAD integration is enabled.  Please open a follow-up issue.

Reply:
https://github.com/Azure/acs-engine/issues/2905